api/drizzle/seeder/generate-products.ts

```diff
@@ -0,0 +1,67 @@
+import { faker } from "@faker-js/faker"
+
+import { db } from "../../src/utils/db"
+import { products } from "../../src/schema/products"
+import { shopsData } from "./generate-shops"
+import { getRandomEntityToLink } from "./helpers/get-random-entity-to-link"
+
+const PRODUCTS_COUNT = 75
+
+type ProductsInput = typeof products.$inferInsert
+
+/**
+ * Generates more mock products for the database.
+ */
+export async function generateProducts() {
+  console.log("generating products...")
+
+  const shops = await db.query.shops.findMany({
+    columns: { id: true, name: true },
+  })
+
+  const categories = await db.query.productCategories.findMany({
+    columns: { id: true, shopId: true },
+  })
+
+  for (let i = 0; i < PRODUCTS_COUNT; i++) {
+    const category = getRandomEntityToLink({
+      primaryEntityIndex: i,
+      entitiesToLink: categories,
+    })
```

Comment:
This helper is tricky, i feel it needs a better name.
The thing is that we have ~18 categories and 75 products.
If we take just a random category for a product, there's a chance that not all categories will have at least a single product in it (random...).

So this helper assigns the first N products to the first N categories, and then continues random assigning of remaining products between categories